src/app/shared/components/km-select/component.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, forwardRef, Input, OnInit} from '@angular/core';
+import {FormBuilder, FormGroup, NG_VALIDATORS, NG_VALUE_ACCESSOR, ValidatorFn, Validators} from '@angular/forms';
+import {BaseFormValidator} from '@app/shared/validators/base-form.validator';
+
+export enum KmSelect {
+  Main = 'main',
+}
+
+@Component({
+  selector: 'km-select',
+  templateUrl: './template.html',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => KmSelectComponent),
+      multi: true,
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => KmSelectComponent),
+      multi: true,
+    },
+  ],
+})
+export class KmSelectComponent extends BaseFormValidator implements OnInit {
```

Comment:
Shouldn't it implement following interface?
```suggestion
ControlValueAccessor, Validator
```

see following shared component
- ChipListComponent
- ChipAutocompleteComponent

Reply:
im not using any methods from those interface

Reply:
using `BaseFormValidator` is also fine, and is being used in some components too, we should encourage it's use as it will reduce code in each component.
<img width="688" alt="image" src="https://user-images.githubusercontent.com/56511988/184610012-981ef0dc-0e35-4850-a517-d7bf5f21da45.png">

Reply:
`BaseFormValidator` is fine but we need to override `registerOnTouched` instead of subscribing to whole form we need to subscribe to individual control so that when value is changed. Value received on parent should be only that control value instead of `obj` e.g km-autocomplete -> we pluck value using `main?.` we don't want that behaviour

Reply:
`BaseFormValidator` is already implementing `ControlValueAccessor` and `Validator` so if you want to override `registerOnTouched` method you can already do that in this class. (this is not related to later part of the comment)